graph_weather/models/fengwu_ghr/layers.py

```diff
@@ -0,0 +1,273 @@
+from scipy.interpolate import griddata
+from torch_geometric.nn import knn
+from torch_geometric.utils import scatter
+import numpy as np
+from scipy.interpolate import griddata, interpn
+import torch
+from einops import rearrange
+from einops.layers.torch import Rearrange
+from torch import nn
+
+
+# helpers
+
+
+def pair(t):
+    return t if isinstance(t, tuple) else (t, t)
+
+
+def knn_interpolate(x: torch.Tensor, pos_x: torch.Tensor, pos_y: torch.Tensor,
+                    k: int = 3, num_workers: int = 1):
+    with torch.no_grad():
+        assign_index = knn(pos_x, pos_y, k,
+                           num_workers=num_workers)
+        y_idx, x_idx = assign_index[0], assign_index[1]
+        diff = pos_x[x_idx] - pos_y[y_idx]
+        squared_distance = (diff * diff).sum(dim=-1, keepdim=True)
+        weights = 1.0 / torch.clamp(squared_distance, min=1e-16)
+
+    # print((x[x_idx]*weights).shape)
```

Comment:
I would remove commented out code like this when you finish with the changes, just to keep it a bit cleaner.